github3/gists/file.py

```diff
@@ -1,51 +1,87 @@
 # -*- coding: utf-8 -*-
-"""
-github3.gists.file
-------------------
-
-Module containing the logic for the GistFile object.
-"""
+"""Module containing the GistFile object."""
 from __future__ import unicode_literals
 
 from ..models import GitHubCore
 
 
-class GistFile(GitHubCore):
+class _GistFile(GitHubCore):
+    """Base for GistFile classes."""
+
+    def _update_attributes(self, gistfile):
+        self.raw_url = gistfile['raw_url']
+        self.filename = gistfile['filename']
+        self.language = gistfile['language']
+        self.size = gistfile['size']
+        self.type = gistfile['type']
+
+    def _repr(self):
+        return '<{s.class_name} [{s.name}]>'.format(s=self)
+
+    def content(self):
+        """Retrieve contents of file from key 'raw_url'.
+
+        :returns: unaltered, untruncated contents of file.
+        :rtype: bytes
+        """
+        resp = self._get(self.raw_url)
+        if self._boolean(resp, 200, 404):
+            return resp.content
+        return None
+
 
+class ShortGistFile(_GistFile):
     """This represents the file object returned by interacting with gists.
 
-    It stores the raw url of the file, the file name, language, size and
-    content.
+    The object has the following attributes as returned by the API for a Gist:
+
+    .. attribute:: raw_url
+
+        This URL provides access to the complete, untruncated content of the
+        file represented by this object.
+
+    .. attribute:: filename
+
+        The string for the filename.
 
+    .. attribute:: language
+
+        The GitHub detected language for the file, e.g., Erlang, Python, text.
+
+    .. attribute:: type
+
+        The mime-type of the file. Related to :attr:`language`.
+
+    .. attribute:: size
+
+        The file size in bytes.
     """
 
-    def _update_attributes(self, gistfile):
-        #: The raw URL for the file at GitHub.
-        self.raw_url = self._get_attribute(gistfile, 'raw_url')
+    class_name = 'ShortGistFile'
 
-        #: The name of the file.
-        self.filename = self._get_attribute(gistfile, 'filename')
 
-        #: The name of the file.
-        self.name = self._get_attribute(gistfile, 'filename')
```

Comment:
This attribute `name` is removed, however it's relied upon in the `_repr` from the parent class. I think that may cause a traceback when trying to print the object.